src/adapter/z-stack/znp/znp.ts

```diff
@@ -159,7 +228,11 @@ class Znp extends events.EventEmitter {
     }
 
     public static async isValidPath(path: string): Promise<boolean> {
-        return SerialPortUtils.is(path, autoDetectDefinitions);
+ 
+        if (SocketPortUtils.isTcp(path)) {
+            return SocketPortUtils.isValidTcpPath(path);
```

Comment:
I have one concern here, for TCP it checks wether this is a vaild TCP path instead of being a Zstack adapter. This will give problems when e.g. the Deconz adapter is integrated (when also used via TCP). I see 2 possible solutions here:

- Not sure if this is possible, but retrieve the vendor ID, similar to what is done with the autodetectdefinitions.
- Return false here. it will default to the zstack adapter anyway (https://github.com/Koenkk/zigbee-herdsman/blob/master/src/adapter/adapter.ts#L66). When the deconz adapter is introduced we should then allow to specify the user what adapter is used. But that is for later.

Reply:
Retrieving Vendor ID is not possible since sockets don't have such. I'll try the 'false' approach. Do you mean to return false whether 'tcp' is the path prefix?

Reply:
Yes, as we cannot determine if it is a valid path is should return false.

Reply:
@Koenkk, i have changed that and added the neccessary test cases. It works as expected. Can you check again? Thank.